Mlem/Views/Shared/Composer/PostDetailEditorView.swift

```diff
@@ -224,5 +199,74 @@ struct PostDetailEditorView: View {
         }
         .navigationBarColor()
         .navigationBarTitleDisplayMode(.inline)
+        .photosPicker(isPresented: $showingPhotosPicker, selection: $imageSelection, matching: .images)
+        .onChange(of: imageSelection) { newValue in
+            if newValue != nil {
+                uploadImage()
```

Comment:
Adding logic to delete the post would be better, I think. That way the user can start writing the post body while it's uploading. I'll add that 👍

Reply:
Ok, I've added this. If you press the `X` when the file is uploading, the upload task is cancelled. If you press it once the file has finished uploading, it sends a delete request.

Reply:
A bit late to this with some thoughts:
- If that delete request fails, how do we make sure users have control of what's happening?
- If we upload before users "truly" commit (I take that to mean tapping on post button) to posting, we're trusting that the remote code doesn't do anything nefarious or doesn't keep cached/archived versions around?

Reply:
One more thought:
- Should we add an alert after an image is picked that says we'll immediately start uploading that image (a show once only alert)? I'm probably in the minority here, but having data leave the device before users get a chance to commit to it might be breaking some sort of trust? Right up to selecting an image, my mental model is basically: "Ok, lets put this into the composer" (like an email) and not "alright internet, have my data now".

Reply:
1) We don't, right now. The delete request has no response, but obviously returns a status code which we can use to determine whether the deletion was successful. I'm not sure in what scenario you would be able to upload an image but not delete it.  It can probably happen, in which case the user's image will not be deleted. We could possibly show an alert in this case.

2) Yes, pretty much. The safer method would be to upload the image only when the user clicks "post". This slows down the post creation flow though. My problem with this relates to how we'd handle inline image uploading - when we get around to implementing those (which I may look into next week), we'll _need_ the image link before the user clicks 'post'. Of course, we could add placeholder text or something, but I feel like that's over-complicating it.

Reply:
> Should we add an alert after an image is picked that says we'll immediately start uploading that image (a show once only alert)?

Yeah, we could do. I'd rather add an upload confirmation than have it upload the image only when you hit 'post'

Reply:
1. Yea, it's a corner case, I'm just thinking very slow connection (for whatever reason) or server state gone wrong.
2. Fair, it's also a corner case (I mean, most services do this anyway).

I think I'd advocate for a showing one-time alert right between the "user taps photo in picker" action and "begin uploading image" action saying something to the effect "we'll begin uploading immediately".

Reply:
Yeah I agree, I'll add that tomorrow 👍

Reply:
Added 👍

Reply:
Nice! Mind if I go in and make a minor tweak so that we only ask permission once unless user turns on the "Ask" toggle?
- Toggle would default to `off` instead of `on`.
<img width="599" alt="Screenshot 2023-09-30 at 6 59 11 PM" src="https://github.com/mlemgroup/mlem/assets/2549615/e708cbe0-c5c0-4c85-9c25-5080505ae4f3">

Reply:
I don't mind whether it's off or on by default. Feel free to change it 👍

Although, changing that line of code won't make a difference I don't think. By that point, the AppStorage value has already been set by PostDetailEditorView.

Reply:
Added change here: [324feda](https://github.com/mlemgroup/mlem/pull/652/commits/324feda2db846528d443df3ecdf5ff4793612b8f)